lua_modules/file_lfs/make_resource.lua

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/lua
+
+--------------------------------------------------------------------------------
+-- Script to be used on PC to build resource.lua file
+-- Parameters: [-o outputfile] file1 [file2] ...
+-- Example: ./make_resource resource/*
+--   creates resource.lua file with all files in resource directory
+--------------------------------------------------------------------------------
+
+local OUT = "resource.lua"
+
+local function readfile(file)
+  local f = io.open(file, "rb")
+  if f then
+    local lines = f:read("*all")
+    f:close()
+    return lines
+  end
+end
+
+-- tests the functions above
+print(string.format("make_resource script - %d parameter(s)", #arg))
+print("parameters: [-o outputfile] file1 [file2] ...")
+
+local larg = {}
+local outpar = false
+for i, a in pairs(arg) do
+  if i>0 then
+    if outpar then
+      OUT = a
+      outpar = false
+    else
+      if a == "-o" then
+        outpar = true
+      else
+        table.insert(larg, a)
+      end
+    end
+  end
+end
+print(string.format("output set to: %s", OUT))
+
+local res = io.open(OUT, "w")
+res:write("--  luacheck: max line length no\nlocal arg = ...\n")
+res:write(('local table_insert=table.insert\n'):format(inp, content))
+res:write(('local filelist={}\n\n'):format(inp, content))
+res:close(file)
+
+for _, a in pairs(larg) do
+  local inp = string.match(a, ".*[/](.*)")
+  if not inp then inp = a end
+  local content = readfile(a)
+  print(string.format("# processing %s", inp))
+
+  if content then
+    res = io.open(OUT, "a")
+    res:write(('table_insert(filelist, "%s")\n'):format(inp, content))
```

Comment:
This implementation will work but create the list in memory each time a resource is requested.
It would be better to build up the { "name1","name2"} string in this script and then generate code like
`if arg == nil then return { "name1","name2"} end`

Reply:
You're right. I was thinking about the possibility of copying blocks of the resource file into another one. Blocks as
```Lua
table_insert(filelist, "index.html")
if arg == "index.html" then return "..." end
```
But your the costs outweight the result. Adjusted as you propose.